core/jazz_environments/index.js

```diff
@@ -74,7 +76,28 @@ var handler = (event, context, cb) => {
                     service = event.query.service.toLowerCase();
                     domain = event.query.domain.toLowerCase();
 
-                    var update_environment_payload = Object.assign({}, event.body);
+                    var update_environment_payload = event.body;
+
+                    if(update_environment_payload.deployment_descriptor) { // If deployment descriptor is present then validate
+                        try {
+                            const outstandingResources = deployDescrValidatorMod.validateResources(update_environment_payload.deployment_descriptor);
+                            if(outstandingResources.length) { 
+                            reject({result: 'inputError', message: `Invalid deployment_descriptor. The resource types not allowed ${outstandingResources}`});
+                            } else {
+                            const outstandingEvents = deployDescrValidatorMod.validateEvents(update_environment_payload.deployment_descriptor);
+                            if(outstandingEvents.length) { // some events that are not allowed were found so let's reject the request
+                                reject({result: 'inputError', message: `Invalid deployment_descriptor. The event types not allowed ${outstandingEvents}`});
+                            } else {
+                                const outstandingActions = deployDescrValidatorMod.validateActions(update_environment_payload.deployment_descriptor);
+                                if(outstandingActions.length) {
+                                    reject({result: 'inputError', message: `Invalid deployment_descriptor. The action types not allowed ${outstandingActions}`});
+                                }
+                            }
+                            }
+                        } catch(e) {
+                        reject({result: 'inputError', message: `Invalid deployment_descriptor format. Nested exception is ${e}`});
+                        }
+                    }
```

Comment:
Test cases are written for these changes?

Reply:
Updated test case